crates/engine/tree/src/tree/payload_processor/multiproof.rs

```diff
@@ -170,11 +170,6 @@ impl ProofSequencer {
         while let Some(pending) = self.pending_proofs.remove(&current_sequence) {
             consecutive_proofs.push(pending);
             current_sequence += 1;
-
```

Comment:
The while let with remove already stops when the next sequence is missing, so the behaviour should be the same.